include/revng/PipelineC/Tracing/Trace.h

```diff
@@ -0,0 +1,229 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "llvm/Support/Error.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+
+namespace detail {
+
+template<typename T>
+inline T toInt(const llvm::StringRef StrInt) {
+  T Result;
+  revng_assert(!StrInt.getAsInteger(10, Result));
+  return Result;
+}
+
+} // namespace detail
+
+namespace revng::tracing {
+
+enum ArgumentState { Undefined, Scalar, Sequence };
+
+struct Argument {
+private:
+  ArgumentState State = Undefined;
+
+public:
+  std::string Scalar;
+  std::vector<std::string> Sequence;
```

Comment:
Make these private and add an accessor asserting `isScalar` and `isSequence` respectively.

Reply:
Implemented